pyanaconda/network.py

```diff
@@ -1530,9 +1537,11 @@ def networkInitialize(ksdata):
         logIfcfgFiles(msg)
 
     # initialize ksdata hostname
-    if ksdata.network.hostname is None:
-        hostname = hostname_from_cmdline(flags.cmdline) or DEFAULT_HOSTNAME
-        update_hostname_data(ksdata, hostname)
+    network_proxy = DBus.get_proxy(MODULE_NETWORK_NAME, MODULE_NETWORK_PATH)
+    if network_proxy.Hostname == DEFAULT_HOSTNAME:
+        bootopts_hostname = hostname_from_cmdline(flags.cmdline)
```

Comment:
I don't think modules should parse cmdline. It should be done by anaconda (boss) which later sets up respective module property, which effectively happens two lines below.

Reply:
If it will be handled by Boss than yes, but I would take caution on doing some complex logic based on the cmdline args. It could require multiple modules to do checks or setup which could result in nasty for synchronization issues.